pip/worker/pypi.go

```diff
@@ -82,32 +84,36 @@ var HashAlgoPickOrder []meta.HashAlgorithm = []meta.HashAlgorithm{
 	meta.HashAlgoMD2,
 }
 
-func makeGetRequest(packageJSONURL string) (*http.Response, error) {
-	url := "https://" + packageJSONURL
-
-	request, _ := http.NewRequest("GET", url, nil)
-	request.Header.Set("Accept", "application/json")
+type pypiPackageDataFactory struct {
+	client *helper.Client
+}
 
-	client := &http.Client{}
-	response, err := client.Do(request)
-	if err != nil {
-		return nil, err
-	}
+type PypiPackageDataFactory interface {
+	GetPackageData(packageJSONURL string) (PypiPackageData, error)
```

Comment:
I'm trying to understand why we need a new interface type. Could you explain?

Reply:
The interface was introduced so that all receiver functions could be mocked in tests.
For instance, `BuildModule` fetches package data using `GetPackageData` and for writing unit tests for `BuildModule` it might be necessary to mock the call to `GetPackageData`.